### PR TITLE
Add Back Button For /product-details page

### LINKED ISSUE #238: [Bug] No way to close overlays for Materials & Fit and Care Instructions displays
# Overview

As a user, when I am browsing the product details page, ie. [/product/dark-deploy-tee](https://store.gatsbyjs.org/product/dark-deploy-tee), If I click on either the *Materials & Fit* and *Care Instructions* Links a full page overlay is displayed. It takes up the full screen, but there is no way to dismiss or close the overlay to get back to the content you were looking at. 

If you are going to stick with this as a modal (I think I saw some discussion of moving to a separate page, #89 ) then there needs to be an **X** close button.

### PR DESCRIPTION
**Pull request by isaac-martin**
Closes #238
- Created a new `BackButton` component
- Added function that will either go back to the product page - specified by URL param `?fromProduct` or return to the homepage on the off chance someone stumbles across that page
- Positioned the button to be sticky because due to the #hash deep linking you sometimes would not see the button when you landed on the page

### CODE REVIEW

**Comment by isaac-martin** [comment]
Actually @jlengstorf Theres some weird behaviour with the sidebar open / closing and the overflow-x. Will need to sort out something else for that sticky button. Keep 👀 open for changes.

**Comment by isaac-martin** [comment]
After playing around with this - maybe the UX of the button isn't that great when you are not always being placed at the top of the document?

Having it sticky works from a UX experience but its a bit clunky code wise.
- We would need to set a height on the `<TextContainer>` component and overflow to scroll. This works, but then you get some weird experience with multiple scrollbars. I guess we could then hide the ones the `<TextContainer>` generates because we would still have the Window ones?

Thoughts?

File: src/components/ProductDetails/BackButton.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    rgba(255, 255, 255, 0.25) 88%,
+    rgba(255, 255, 255, 0.25) 94%,
+    rgba(255, 255, 255, 0) 94%,
+    rgba(255, 255, 255, 0) 100%
+  );
+  bottom: 0;
+  left: 0;
+  padding: ${spacing.xl}px ${spacing.md}px ${spacing.md}px 0;
+  position: sticky;
+  top: 40px;
```
Comment (jlengstorf) [inline src/components/ProductDetails/BackButton.js]: I think this'll need to be `top: 39px` or so. I'm seeing a gap on Chrome.

Comment (isaac-martin) [reply src/components/ProductDetails/BackButton.js]: Changed this to `top: 38px`

**Comment by isaac-martin** [comment]
@jlengstorf there was at some point weird behaviour with scrollbars on my machine, but I can't rectify. 
Fixed up those issues you mentioned.

**Pull request merged**